cedar-drt/fuzz/src/schemas.rs

```diff
@@ -100,7 +106,184 @@ fn namespace_equivalence<N: Clone + PartialEq + std::fmt::Debug + std::fmt::Disp
     }
 }
 
-fn action_type_equivalence<N: PartialEq + std::fmt::Debug + std::fmt::Display>(
+type EntityData<N> = HashMap<UnreservedId, EntityType<N>>;
+
+fn entity_types_equivalence<N: Clone + PartialEq + Debug + Display + TypeName>(
+    lhs: EntityData<N>,
+    rhs: EntityData<N>,
+) -> Result<(), String> {
+    if lhs.len() == rhs.len() {
+        let errors = lhs
+            .into_iter()
+            .filter_map(|lhs| entity_type_equivalence(lhs, &rhs).err())
+            .collect::<Vec<_>>();
+        if errors.is_empty() {
+            Ok(())
+        } else {
+            Err(format!(
+                "Found the following entity type mismatches: {}",
+                errors.into_iter().join("\n")
+            ))
+        }
+    } else {
+        let lhs_keys: HashSet<_> = lhs.keys().collect();
+        let rhs_keys: HashSet<_> = rhs.keys().collect();
+        let missing_keys = lhs_keys.symmetric_difference(&rhs_keys).join(", ");
+        Err(format!("Missing keys: {missing_keys}"))
+    }
+}
+
+fn entity_type_equivalence<N: Clone + PartialEq + Debug + Display + TypeName>(
+    (name, lhs_type): (UnreservedId, EntityType<N>),
+    rhs: &EntityData<N>,
+) -> Result<(), String> {
+    let rhs_type = rhs
+        .get(&name)
+        .ok_or_else(|| format!("Type `{name}` was missing from right-hand-size"))?;
+
+    if lhs_type.member_of_types != rhs_type.member_of_types {
+        Err(format!(
+            "For `{name}`: lhs and rhs membership are not equal. LHS: [{}], RHS: [{}].",
+            lhs_type
+                .member_of_types
+                .into_iter()
+                .map(|id| id.to_string())
+                .join(","),
+            rhs_type
+                .member_of_types
+                .iter()
+                .map(|id| id.to_string())
+                .join(",")
+        ))
+    } else if shape_equiv(&lhs_type.shape.0, &rhs_type.shape.0) {
+        Ok(())
+    } else {
+        Err(format!("`{name}` has mismatched types"))
+    }
+}
+
+fn shape_equiv<N: Clone + PartialEq + TypeName>(lhs: &Type<N>, rhs: &Type<N>) -> bool {
+    match (lhs, rhs) {
+        (Type::Type(lhs), Type::Type(rhs)) => type_var_equiv(lhs, rhs),
+        (Type::CommonTypeRef { type_name: lhs }, Type::CommonTypeRef { type_name: rhs }) => {
+            lhs == rhs
+        }
+        _ => false,
+    }
+}
+
+/// Type Variant equivalence. See the arms of each match for details
+fn type_var_equiv<N: Clone + PartialEq + TypeName>(
```

Comment:
naming nit: `type_var` reads like `type_variable` to me, so I'd rather spell this out as `type_variant_equiv`